src/modules/orbitdb/abstract.js

```diff
@@ -21,27 +40,26 @@ class ChluAbstractIndex {
             // TODO: handle errors
             if (item.payload.version === this._version) {
                 // TODO: check update validity, RR validity if possible
-                if (item.payload.op === operations.ADD_REVIEW_RECORD && IPFSUtils.isValidMultihash(item.payload.multihash)) {
-                    await this.addOriginalReviewRecord({
-                        multihash: item.payload.multihash,
-                        subjectDidId: item.payload.subjectDidId || item.payload.didId, // Retrocompatibility
-                        authorDidId: item.payload.authorDidId,
-                        bitcoinTransactionHash: item.payload.bitcoinTransactionHash
-                    });
-                } else if (item.payload.op === operations.UPDATE_REVIEW_RECORD) {
-                    if (IPFSUtils.isValidMultihash(item.payload.multihash) && IPFSUtils.isValidMultihash(item.payload.previousVersionMultihash)) {
-                        // TODO: check that whatever is being updated is in the DB
-                        // the multihash to be updated might already be an update. Be careful!
-                        const from = await this.getLatestReviewRecordUpdate(item.payload.previousVersionMultihash);
-                        await this.addReviewRecordUpdate({
-                            fromMultihash: from,
-                            toMultihash: item.payload.multihash
+                if (item.payload.op === operations.ADD_REVIEW_RECORD) {
+                    try {
+                        const reviewRecord = await this.getAndValidateReviewRecordContent(item.payload.multihash)
+                        // TODO retry if validation failed 
+                        await this.addReviewRecord({
+                            multihash: item.payload.multihash,
+                            reviewRecord,
+                            bitcoinTransactionHash: item.payload.bitcoinTransactionHash || null
                         });
+                    } catch (error) {
+                        this.chluIpfs.logger.error(`Error while updating ChluDB Index: ${error.message}`)
```

Comment:
The review won't be in the Index, this means that we won't have the metadata (payment information, the transaction hash) so it will never be valid again unless the process is restarted and this time it goes through validation at this step.

The review is not in the index so it won't show up in the various lists like reviews by author, but if you attempt to read the review directly through readReviewRecord(multihash) then it will try to read it and validate it again

So we are not solving this yet, but I'm open to ideas